### PR TITLE
Kill ResourceGroup

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16288)

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -68,14 +69,18 @@ pub struct HttpState {
     pub hsts_list: RwLock<HstsList>,
     pub cookie_jar: RwLock<CookieStorage>,
     pub auth_cache: RwLock<AuthCache>,
+    pub ssl_client: OpensslClient,
```
Comment (jdm) [inline components/net/http_loader.rs]: Do we actually need to store this at all? It looks like we never use it.

**Review by jdm — DISMISSED** [review]

**Comment by nox** [comment]
I will need the SSL client for wss.

**Comment by nox** [comment]
Borrowing issue, fixed now.

@bors-servo r=jdm

**Pull request merged**